src/main/java/com/icthh/xm/uaa/security/DomainUserDetailsService.java

```diff
@@ -71,8 +71,10 @@ private DomainUserDetails buildDomainUserDetails(String lowerLogin, String tenan
                                         .collect(toList());
 
         // get user role authority
-        SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRoleKey());
-        List<SimpleGrantedAuthority> authorities = Collections.singletonList(authority);
+        List<SimpleGrantedAuthority> authorities = user.getAuthorities()
+            .stream()
+            .map(role -> new SimpleGrantedAuthority(role))
```

Comment:
use method reference here instead of lambda.

Reply:
done